### PR TITLE
adds extensions meta in geojson output

### PR DESCRIPTION
**Pull request by nerik**
Hi, thank you for that very useful lib 👍 

We have a backend script that converts shapefiles using a different lib, whose requirement is for a zipped shapefile to contain shp, shx, and dbf components (which seems to be the official recommendation, see https://www.loc.gov/preservation/digital/formats/fdd/fdd000280.shtml).

shpjs has a specific strategy to deal with the absence of a dbf file, and it doesn't seem to require an shx file but can work with an sbx file. Therefore, I can't just throw an error without introducing a breaking change - and without breaking tests.

So I suggest having the output GeoJSON containing an `extensions` meta field (similar as `fileName`) to allow users to deal with their own shapefile requirements should they want to.

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
so the thing about an shx is that it's only needed for doing look ups within the shp file, since we're just reading straight through the shp file we don't actually care if it's there or not.

So two issues:

1. I don't totally get why you'd need this?
2. some of the file extensions actually have two dots in them notably `.shp.xml`, and then you could also have a just regular `.xml` file in there too.
3. what you've written makes a list of all the extentions of all the files in the zipfile and then puts them on everyfile, so if you have two shapefiles each with .shp, .dbf and .shx files in one zip, you'd have an array on each with .shp, .dbf, .shx , .shp, .dbf and .shx in it.

**Comment by nerik** [comment]
Thanks for reviewing. 

Our use case is that we use shapefile-js on the frontend to read feature properties, but the shapefile is then uploaded to a backend service that does its own shapefile parsing, which in our setup needs shp, shx, dbf, as well as prj files. This PR allows us to intercept a shapefile that's not going to work with our backend before uploading.

I guess it's a bit of a convoluted scenario, but I imagined adding some metadata might be useful to someone else. 
Let me know if you could be interested in merging this, I can easily fix the two remaining issues: 
2. just report whatever comes after the first dot?
3. yep, good catch, I was distracted 🙄

**Comment by calvinmetcalf** [comment]
you can find the basename by looking at the basename.shp and then grabbing everything after `basename.`

**Comment by nerik** [comment]
@calvinmetcalf I fixed the two issues you reported and added a test.

**Comment by nerik** [comment]
@calvinmetcalf Hey there, sorry to bother you - do you plan to merge this or should we use my fork? Thanks

**Comment by calvinmetcalf** [comment]
sorry was on vacation all last week, I'll take a look this week

File: lib/index.js [hunk 1]
```diff
@@ -98,11 +99,19 @@ shp.parseZip = async function (buffer, whiteList) {
       parsed = zip[name];
       parsed.fileName = name;
     } else {
+      const extensions = [];
+      for (key in zip) {
+        const extension = key.replace(name, '');
+        if (extension !== key) {
```
Comment (calvinmetcalf) [inline lib/index.js:105]: i think you also need to do a check to make sure that extension starts with a '.'

**Comment by calvinmetcalf** [comment]
closing this, I dropped the ball on merging this and I've changed way to much to merge it in as is

**Pull request closed without merging**